scripts/gen_syscall_header.py

```diff
@@ -95,6 +95,11 @@ def gen_defines_inner(ret, argc, kernel_only=False, user_only=False):
         sys.stdout.write("\t\t} else {")
         newline()
 
+        # Prevent memory access issues if the implementation function gets
+        # inlined
+        sys.stdout.write("\t\t\tcompiler_barrier();");
```

Comment:
There's an `__attribute__((noinline))` that could be used for this purpose as well. Not sure which one is better.

Reply:
I don't think we want to do that as that would effectively make all those implementation functions which have been defined as inline, not inline.

Unless you have found a way to attach the noinline declaration to the call site and not the function definition?